aio_pika/queue.py

```diff
@@ -63,7 +66,10 @@ def declare(self, timeout: int = None, passive: bool = False) -> asyncio.Future:
         )
 
         def on_queue_declared(result):
-            self.name = result.result().method.queue
+            method = result.result().method
+            self.name = method.queue
+            self.message_count = method.message_count
+            self.consumer_count = method.consumer_count
```

Comment:
Yes, it's server-side state that gets returned with every `queue-ok` response from the AMQP server.

It does seem a bit odd putting it on `Queue` (since it will be out-of-date as soon as it's used), but it's the easiest thing to do.

An alternative would be to expose the underlying `Queue.declare` future somehow. Obviously it would be possible to 'manually' call `Queue.declare` but then you also have to manually take care of all the [setup in Channel](https://github.com/mosquito/aio-pika/blob/master/aio_pika/channel.py#L195).